src/Microsoft.ML.DnnImageFeaturizer.AlexNet/AlexNet.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using System.IO;
+using System.Reflection;
+
+namespace Microsoft.ML.Transforms
+{
+    /// <summary>
+    /// This is an extension method to be used with the <see cref="DnnImageFeaturizerEstimator"/> in order to use a pretrained AlexNet model.
+    /// The NuGet containing this extension is also guaranteed to include the binary model file. Note that when building the project
+    /// containing this extension method, the corresponding binary model will be downloaded from the CDN at
+    /// https://express-tlcresources.azureedge.net/image/AlexNetPrepOnnx/AlexNetPreprocess.onnx and
+    /// https://express-tlcresources.azureedge.net/image/AlexNetOnnx/AlexNet.onnx  and placed into the local app directory
+    /// folder under mlnet-resources.
+    /// </summary>
+    public static class AlexNetExtension
+    {
+        /// <summary>
+        /// If including this through a NuGet, the location of the model will be the same as of this file. This looks for the model there.
+        /// This should be the default way to use AlexNet if importing the model from a NuGet.
+        /// </summary>
+        public static EstimatorChain<OnnxTransform> AlexNet(this DnnImageModelSelector dnnModelContext, IHostEnvironment env, string input, string output)
+        {
+            var modelChain = new EstimatorChain<OnnxTransform>();
+            var tempCol = "onnxDnnPrep";
+            var execDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+
```

Comment:
```suggestion
// There are two estimators created below. The first one is for image preprocessing and the second one is the actual DNN model.
```

Reply:
I think in this case, the names of the models are clear enough that this is not necessary.

---
In reply to: [232437834](https://github.com/dotnet/machinelearning/pull/1447#discussion_r232437834) [](ancestors = 232437834)